File: 658-find-k-closest-elements/658-find-k-closest-elements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
      answer=[]
      st=0
      end=len(arr)-1
      while end-st >=k:
        if abs(arr[st]-x) > abs(arr[end]-x):
          st+=1
        else:
          end-=1
      
      for i in range(st,end+1):
        answer.append(arr[i]);
      
      
      return answer;

File: 658-find-k-closest-elements/658-find-k-closest-elements.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
     
      List<Integer> answer= new ArrayList<>();
      int st=0;
      int end=arr.length-1;
      while(end-st >=k){
        if(Math.abs(arr[st]-x) > Math.abs(arr[end]-x))
           st++;
        else
           end--;
      }
      
      
      for(int i=st;i<=end;i++){
        answer.add(arr[i]);
      }
      return answer;
    }
}